python/box.tests/reports/git-history-plot.svg

```diff
@@ -213,9 +213,9 @@ text {
         <line x1="0.0" y1="0.0" x2="0.0" y2="451.0" stroke-width="1.0" stroke="rgb(0,0,0)" stroke-opacity="1.0">
         </line>
       </g>
-      <g transform="translate(42.5 0.0 ) " clip-path="url(#CzUoYF)" clip-bounds-jfx="[rect (0.0, 0.0), (747.15, 451.0)]">
```

Comment:
It would be really cool to remove these non-deterministic IDs. They will trigger the automated commit even if we fix a typo in the README :) I know it's technically not a part of the story, but now with automated commits this issue will become more apparent - so far I just discarded such non-meaningful changes locally.

I propose a simple regex-based approach, so that we don't go into parsing XML/SVG. It can be implemented in our `kts` script:

* find all patterns like `"url\(#\w{6}\)"` and `id="\w{6}"`, extract ID from each
* for each found pattern and ID:
  * if it **wasn't** already seen, assign a deterministic ID for this particular random ID, e.g.  create a mapping from `4EUPyR` to `id1`, and use it
  * if it **was** already seen, look up its deterministic ID, and use it

Thanks to this, even if lets-plot generates a new SVG which only differs in these IDs and no real differences in data, the above random-to-deterministic-IDs algo will "normalize" it and lead to in fact no change in the SVG.

Reply:
Will do.

Do you think we can report it to https://github.com/JetBrains/lets-plot/issues? I think the lib should support creating deterministic results itself. But maybe I'm not seeing something why it can't be implemented from the lib's side...

Reply:
Yes, definitely it should be fixed in Lets-plot, what I proposed is basically a workaround. Best if we didn't have to apply it :)

Reply:
Filed: https://github.com/JetBrains/lets-plot/issues/492. Will apply the workaround you've suggested here.

Reply:
If you think the workaround can be improved, please suggest.